route_test.go

```diff
@@ -45,6 +46,129 @@ func TestFallback(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, "not found", string(body))
 }
+func TestListen(t *testing.T) {
+	var (
+		err        error
+		mockConfig *configmocks.Config
+		route      *Route
+	)
+
+	tests := []struct {
+		name        string
+		setup       func(host string, port string) error
+		host        string
+		port        string
+		expectError error
+	}{
+		{
+			name: "success listen",
+			setup: func(host string, port string) error {
+				go func() {
+					l, err := net.Listen("tcp", host)
+					assert.Nil(t, err)
+					assert.Nil(t, route.Listen(l))
+				}()
+				time.Sleep(1 * time.Second)
+
+				return errors.New("error")
+			},
+			host: "127.0.0.1:3100",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mockConfig = configmocks.NewConfig(t)
+			mockConfig.EXPECT().GetBool("http.drivers.fiber.prefork", false).Return(false).Once()
+			mockConfig.EXPECT().GetInt("http.drivers.fiber.body_limit", 4096).Return(4096).Once()
+			mockConfig.EXPECT().GetInt("http.drivers.fiber.header_limit", 4096).Return(4096).Once()
+
+			route, err = NewRoute(mockConfig, nil)
+			assert.Nil(t, err)
+			route.Get("/", func(ctx contractshttp.Context) contractshttp.Response {
+				return ctx.Response().Json(200, contractshttp.Json{
+					"Hello": "Goravel",
+				})
+			})
+			if err := test.setup(test.host, test.port); err == nil {
+				time.Sleep(1 * time.Second)
+				hostUrl := "http://" + test.host
+				if test.port != "" {
+					hostUrl = hostUrl + ":" + test.port
+				}
+				resp, err := http.Get(hostUrl)
+				assert.Nil(t, err)
+				defer resp.Body.Close()
+
+				body, err := io.ReadAll(resp.Body)
+				assert.Nil(t, err)
+				assert.Equal(t, "{\"Hello\":\"Goravel\"}", string(body))
+			}
+			mockConfig.AssertExpectations(t)
```

Comment:
Since we are using `configmocks.NewConfig(t)`, the `AssertExpectations` method can be removed.